src/include/sof/math/exp_fcn_hifi4.h

```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOF_MATH_EXPONENTIAL_HIFI4_H__
+#define __SOF_MATH_EXPONENTIAL_HIFI4_H__
+
+#include <stdint.h>
+#include <stddef.h>
+
+#undef EXPONENTIAL_GENERIC
+#undef EXPONENTIAL_HIFI4
```

Comment:
How could these be defined before any SOF `#include`?

Stop undef-ining things "just in case"; we want the pre-processor to catch multiple definitions and fail.